HABITAT_2_TESTING_GUIDE.md

```diff
@@ -0,0 +1,429 @@
+pkg_version="1.0.0"
+pkg_maintainer="Test User <test@company.com>"
+pkg_license=('Apache-2.0')
+pkg_description="Test service for Habitat 2.0 - Cross Platform"
+
+# Cross-platform dependency
+pkg_deps=(
+  core/python
+)
+
+# Platform-agnostic service using Python HTTP server
+pkg_svc_run="python -m http.server 8080"
+
+do_build() {
+  return 0
+}
+
+do_install() {
+  return 0
+}
+EOF
+```
+
+#### 2. Set env
+
+```bash
+export HAB_ORIGIN=test
+
+# Point to your on-prem builder
+export HAB_BLDR_URL=https://your-builder.example.com
+
+# Set authentication token
+export HAB_AUTH_TOKEN=<ON_PREM_BUILDER_TOKEN>
+```
+
+#### 3. Build package
+
+```bash
+# One time command to generate the public key at ~/.hab/cache/keys
+sudo hab origin key generate test
+
+
+# Build using Habitat 2.0 with base channel
+sudo -E  hab pkg build .
+# Make a note of the hart file generated at the end of the successful build.
+```
+
+#### 4. Create Origin at the On Prem Builder (If not already done)
+
+Custom Builder instances don't auto-create origins. 
+If the origin doesn't already exist, you need to manually create the origin and upload your keys first.
+
+```bash
+# 1. Create the origin on your custom Builder
+sudo -E hab origin create test
+
+# 3. Upload your public origin key
+sudo -E hab origin key upload test
+```
+
+#### 5. Upload to on prem bldr
+
+```bash
+sudo -E hab pkg upload <HART_FILE>
+# Make a note of the CUSTOM_PKG_IDENT that gets uploaded.
+# By default , the package gets uploaded onto the unstable channel.
+```
+
+### On a environment with supervisor running (Production Supervisor Environment with Habitat 2.0)
+
+#### Update the systemd service to connect to on prem builder
+
+Create the hab user:
+
+```bash
+sudo groupadd hab
+sudo useradd -g hab hab
+```
+
+Create the systemd service file:
+
+```bash
+sudo tee /etc/systemd/system/hab-sup.service > /dev/null <<EOF
+[Unit]
+Description=The Habitat Supervisor
+After=network-online.target
+Wants=network-online.target
+
+[Service]
+Type=simple
+ExecStart=$(which hab) sup run
+Restart=on-failure
+RestartSec=10
+KillMode=mixed
+KillSignal=SIGINT
+TimeoutStartSec=120
+TimeoutStopSec=60
+Environment=HAB_BLDR_URL=https://your-builder.example.com
+
+[Install]
+WantedBy=multi-user.target
+EOF
+```
+
+Enable and start the service:
+
+```bash
+# Reload systemd configuration
+sudo systemctl daemon-reload
+
+# Enable the service
+sudo systemctl enable hab-sup
+
+# Start the service
+sudo systemctl start hab-sup
+sleep 10
+
+# Verify service is running
+sudo systemctl status hab-sup
+
+# Check supervisor is accessible
+sudo hab svc status
+```
+
+#### Verification
+
+- **Linux**: `sudo systemctl status hab-sup` shows active/running
+- **Windows**: `Get-Service habitat` shows running services
+
+#### Load the package
+
+```bash
+# Set the Builder URL
+export HAB_BLDR_URL=https://your-builder.example.com
+
+# Load the service
+sudo -E hab svc load <CUSTOM_PKG_IDENT> --channel <CHANNEL_ON_WHICH_THE_PKG_WAS_UPLOADED>
```

Comment:
we should explicitly specify the `unstable` channel since that is the channel the package above is located in.